subsys/bluetooth/mesh/brg_cfg_cli.c

```diff
@@ -31,47 +191,170 @@ const struct bt_mesh_model_cb _bt_mesh_brg_cfg_cli_cb = {
+	};
+
+	bt_mesh_model_msg_init(&msg, OP_SUBNET_BRIDGE_GET);
+
+	return bt_mesh_msg_ackd_send(cli->model, &ctx, &msg, !status ? NULL : &rsp_ctx);
 }
 
 int bt_mesh_brg_cfg_cli_subnet_bridge_set(uint16_t net_idx, uint16_t addr,
 					  enum bt_mesh_subnet_bridge_state val,
 					  enum bt_mesh_subnet_bridge_state *status)
 {
-	return 0;
+	BT_MESH_MODEL_BUF_DEFINE(msg, OP_SUBNET_BRIDGE_SET, 1);
+	struct bt_mesh_msg_ctx ctx = BT_MESH_MSG_CTX_INIT_DEV(net_idx, addr);
+	const struct bt_mesh_msg_rsp_ctx rsp_ctx = {
+		.ack = &cli->ack_ctx,
+		.op = OP_SUBNET_BRIDGE_STATUS,
+		.user_data = status,
+		.timeout = msg_timeout,
+	};
+
+	bt_mesh_model_msg_init(&msg, OP_SUBNET_BRIDGE_SET);
+	net_buf_simple_add_u8(&msg, (uint8_t)val);
+
+	return bt_mesh_msg_ackd_send(cli->model, &ctx, &msg, !status ? NULL : &rsp_ctx);
 }
 
 int bt_mesh_brg_cfg_cli_bridging_table_size_get(uint16_t net_idx, uint16_t addr, uint16_t *size)
 {
-	return 0;
+	BT_MESH_MODEL_BUF_DEFINE(msg, OP_BRIDGING_TABLE_SIZE_GET, 0);
+	struct bt_mesh_msg_ctx ctx = BT_MESH_MSG_CTX_INIT_DEV(net_idx, addr);
+	const struct bt_mesh_msg_rsp_ctx rsp_ctx = {
+		.ack = &cli->ack_ctx,
+		.op = OP_BRIDGING_TABLE_SIZE_STATUS,
+		.user_data = size,
+		.timeout = msg_timeout,
+	};
+
+	bt_mesh_model_msg_init(&msg, OP_BRIDGING_TABLE_SIZE_GET);
+
+	return bt_mesh_msg_ackd_send(cli->model, &ctx, &msg, !size ? NULL : &rsp_ctx);
 }
 
 int bt_mesh_brg_cfg_cli_bridging_table_add(uint16_t net_idx, uint16_t addr, uint8_t directions,
 					   uint16_t net_idx1, uint16_t net_idx2,
 					   uint16_t addr1, uint16_t addr2,
 					   struct bt_mesh_bridging_table_status *rsp)
 {
-	return 0;
+	BT_MESH_MODEL_BUF_DEFINE(msg, OP_BRIDGING_TABLE_ADD, 8);
+	struct bt_mesh_msg_ctx ctx = BT_MESH_MSG_CTX_INIT_DEV(net_idx, addr);
+	const struct bt_mesh_msg_rsp_ctx rsp_ctx = {
+		.ack = &cli->ack_ctx,
+		.op = OP_BRIDGING_TABLE_STATUS,
+		.user_data = rsp,
+		.timeout = msg_timeout,
+	};
+
+	if (addr1 == addr2) {
+		LOG_ERR("addr1 and addr2 shall have different values.");
+		return -EINVAL;
+	} else if (!BT_MESH_ADDR_IS_UNICAST(addr1)) {
+		LOG_ERR("addr1 shall be a unicast address.");
+		return -EINVAL;
+	} else if (directions == 0x01 &&
+		   (addr2 == BT_MESH_ADDR_UNASSIGNED || addr2 == BT_MESH_ADDR_ALL_NODES)) {
+		LOG_ERR("For direction 0x01: addr2 shall not be unassigned or the all-nodes fixed "
+			"group address.");
+		return -EINVAL;
+	} else if (directions == 0x02 && !BT_MESH_ADDR_IS_UNICAST(addr2)) {
+		LOG_ERR("For direction 0x02: addr2 shall be a unicast address.");
+		return -EINVAL;
+	}
+
+	bt_mesh_model_msg_init(&msg, OP_BRIDGING_TABLE_ADD);
+	net_buf_simple_add_u8(&msg, directions);
+	net_buf_simple_add_le24(&msg, (net_idx1 | (net_idx2 << 12)));
+	net_buf_simple_add_le16(&msg, addr1);
+	net_buf_simple_add_le16(&msg, addr2);
+
+	return bt_mesh_msg_ackd_send(cli->model, &ctx, &msg, !rsp ? NULL : &rsp_ctx);
 }
 
 int bt_mesh_brg_cfg_cli_bridging_table_remove(uint16_t net_idx, uint16_t addr, uint16_t net_idx1,
```

Comment:
Same here.

Reply:
The `bridging_table_entry` struct contains the `direction` field which is not used in removal. Do you still think we should change this?